.github/workflows/default.yaml

```diff
@@ -109,10 +114,30 @@ jobs:
           sudo apt-get --quiet=2 build-dep squid
           sudo apt-get --quiet=2 install linuxdoc-tools libtool-bin
 
+      - name: Install prerequisite MacOS packages
+        if: runner.os == 'macOS'
+        run: |
+          brew install \
+            automake coreutils cppunit gawk \
+            gnu-getopt gnu-sed grep \
+            make openldap cyrus-sasl
+
       - name: Checkout sources
-        uses: actions/checkout@v3
+        uses: actions/checkout@v4
 
-      - run: ./test-builds.sh
+      - name: Run build on Linux
+        if: runner.os == 'Linux'
+        run: ./test-builds.sh --aggressively-use-config-cache ${{ matrix.layer }}
```

Comment:
Please move these out-of-scope changes into a dedicated PR. Since caching had caused a lot of headaches in Jenkins, that dedicated PR should provide data to back the implied "speed up" claims.

Reply:
This option refers to config.cache (used by configure); using it could be harmful only if the system were to change between runs, but since runners get wiped and rebuilt on each run, no harm is possible.
If anything, if the per-layer parallelisation is accepted, this is useless because there is no chance to
reuse the cache

Reply:
While your response raises more red flags, I will pass this opportunity to discuss these out-of-scope changes in this PR. If you are sure these changes are a good idea, please see my change request for the next steps.